File: java/1265.java
// Problem: https://leetcode.com/problems/print-immutable-linked-list-in-reverse/description/

/**
 * // This is the ImmutableListNode's API interface.
 * // You should not implement it, or speculate about its implementation.
 * interface ImmutableListNode {
 *     public void printValue(); // print the value of this node.
 *     public ImmutableListNode getNext(); // return the next node.
 * };
 */

class Solution {
    public void printLinkedListInReverse(ImmutableListNode head) {
        while(head == null) {
            return;
        }
        printLinkedListInReverse(head.getNext());
        head.printValue();
    }
}

File: TypeScript/1265.ts
// https://leetcode.com/problems/print-immutable-linked-list-in-reverse/description/

/**
 * // This is the ImmutableListNode's API interface.
 * // You should not implement it, or speculate about its implementation
 * class ImmutableListNode {
 *      printValue() {}
 *
 *      getNext(): ImmutableListNode {}
 * }
 */

function printLinkedListInReverse(head: ImmutableListNode) {
	if(head == null) return;
    printLinkedListInReverse(head.getNext());
    head.printValue();
};
